src/tiled/variantpropertymanager.cpp

```diff
@@ -126,6 +135,32 @@ QString VariantPropertyManager::valueText(const QtProperty *property) const
         QVariant value = mValues[property].value;
         int typeId = propertyType(property);
 
+        if (typeId == objectIdTypeId()) {
+            ObjectId id = value.value<ObjectId>();
+
+            if (MapDocument *document = DocumentManager::instance()->currentDocument()) {
+                QList<ObjectGroup*> objectGroups = document->map()->objectGroups();
+
+                foreach(const ObjectGroup *group, objectGroups) {
+                    foreach (const MapObject *object, group->objects())
+                    {
+                        if (object->id() == id.id) {
+                            QString label = QStringLiteral("%1: ").arg(QString::number(object->id()));
+                            if (object->name().length())
+                                label.append(object->name());
+                            else
+                                label.append(QStringLiteral("Unnamed object"));
+
+                            if (object->type().length())
+                                label.append(QStringLiteral(" (%1)").arg(object->type()));
+                            return label;
+                        }
+                    }
+                }
+            }
+            return QStringLiteral("%1: Object not found").arg(QString::number(id.id));
```

Comment:
This should be `tr("%1: Object not found").arg...`

Also, I think a special text would be good for when the `id` is 0, which can never refer to a valid object. Maybe just "Unset".